ibis/omniscidb/client.py

```diff
@@ -586,6 +603,14 @@ def _build_ast(self, expr, context):
         result = build_ast(expr, context)
         return result
 
+    def _check_execution_type(self, ipc: bool, gpu_device: int):
```

Comment:
so gpu_device can be None? can you update the annotation

Reply:
`gpu_device` is a required parameter, and it could  be explicitly `None`.